### DISCUSSION TITLE
Synth/Kit Text Search

### DISCUSSION THREAD

**Discussion by hainezee** [SynthstromAudible/DelugeFirmware#2725] [Ideas]
I was wondering if it might be possible to allow for a text character search <<within>> the text string of synths & kits? I.e. when "Load"-"Synth" is pressed, revealing the text keyboard, typing in "bass" would search and return ALL synths whose name contains these characters within the entire name-string. Currently it only searches and returns those beginning with the input characters... This would help find all synth patches including key phrases like "bass" without having to reorganise synths into folders.

**Comment by m0nkmaster** [comment]
I think this is a good idea. I know that we do currently have search and a keyboard does show when browsing - however the search is pretty difficult to understand.

I like the idea of having the search string you've typed being displayed on the screen (top left perhaps) and the search synth/kit/sample uses a fussy search to find results.

**Reply by m0nkmaster** [reply]
Perhaps using [LucenePlusPlus](https://github.com/luceneplusplus/LucenePlusPlus) or [rapidfuzz-cpp](https://github.com/rapidfuzz/rapidfuzz-cpp)
